x-pack/plugins/aiops/server/routes/queries/get_simple_hierarchical_tree.ts

```diff
@@ -236,29 +242,45 @@ export function getSimpleHierarchicalTreeLeaves(
     }
   }
 
+  if (leaves.length === 1 && leaves[0].group.length === 0 && leaves[0].docCount === 0) {
+    return [];
+  }
+
   return leaves;
 }
 
+type FieldValuePairCounts = Record<string, Record<string, number>>;
 /**
- * Analyse duplicate field/value pairs in change point groups.
+ * Get a nested record of field/value pairs with counts
  */
-export function markDuplicates(cpgs: ChangePointGroup[]): ChangePointGroup[] {
-  const fieldValuePairCounts: Record<string, number> = {};
-  cpgs.forEach((cpg) => {
+export function getFieldValuePairCounts(cpgs: ChangePointGroup[]): FieldValuePairCounts {
+  return cpgs.reduce((p, cpg) => {
     cpg.group.forEach((g) => {
-      const str = `${g.fieldName}$$$$${g.fieldValue}`;
-      fieldValuePairCounts[str] = fieldValuePairCounts[str] ? fieldValuePairCounts[str] + 1 : 1;
+      if (p[g.fieldName] === undefined) {
```

Comment:
real nit pick, this code might be nicer to read if you expand `g` out to `{ fieldName, fieldValue }`

Reply:
Fixed in [4d06d86](https://github.com/elastic/kibana/pull/140891/commits/4d06d866e88cb181cda52272c46a933f0473541e).